pkg/controllers/updaterun/validation_integration_test.go

```diff
@@ -0,0 +1,485 @@
+
+		By("Creating the member clusters")
+		for _, cluster := range targetClusters {
+			Expect(k8sClient.Create(ctx, cluster)).To(Succeed())
+		}
+		for _, cluster := range unscheduledCluster {
+			Expect(k8sClient.Create(ctx, cluster)).To(Succeed())
+		}
+
+		By("Creating a bunch of ClusterResourceBindings")
+		for _, binding := range resourceBindings {
+			Expect(k8sClient.Create(ctx, binding)).To(Succeed())
+		}
+
+		By("Creating a clusterStagedUpdateStrategy")
+		Expect(k8sClient.Create(ctx, updateStrategy)).To(Succeed())
+
+		By("Creating a new resource snapshot")
+		Expect(k8sClient.Create(ctx, resourceSnapshot)).To(Succeed())
+
+		By("Creating a new cluster resource override")
+		Expect(k8sClient.Create(ctx, clusterResourceOverride)).To(Succeed())
+	})
+
+	AfterEach(func() {
+		By("Deleting the clusterStagedUpdateRun")
+		Expect(k8sClient.Delete(ctx, updateRun)).Should(Succeed())
+		updateRun = nil
+
+		By("Deleting the clusterResourcePlacement")
+		Expect(k8sClient.Delete(ctx, crp)).Should(SatisfyAny(Succeed(), utils.NotFoundMatcher{}))
+		crp = nil
+
+		By("Deleting the clusterSchedulingPolicySnapshot")
+		Expect(k8sClient.Delete(ctx, policySnapshot)).Should(SatisfyAny(Succeed(), utils.NotFoundMatcher{}))
+		policySnapshot = nil
+
+		By("Deleting the clusterResourceBindings")
+		for _, binding := range resourceBindings {
+			Expect(k8sClient.Delete(ctx, binding)).Should(SatisfyAny(Succeed(), utils.NotFoundMatcher{}))
+		}
+		resourceBindings = nil
+
+		By("Deleting the member clusters")
+		for _, cluster := range targetClusters {
+			Expect(k8sClient.Delete(ctx, cluster)).Should(SatisfyAny(Succeed(), utils.NotFoundMatcher{}))
+		}
+		for _, cluster := range unscheduledCluster {
+			Expect(k8sClient.Delete(ctx, cluster)).Should(SatisfyAny(Succeed(), utils.NotFoundMatcher{}))
+		}
+		targetClusters, unscheduledCluster = nil, nil
+
+		By("Deleting the clusterStagedUpdateStrategy")
+		Expect(k8sClient.Delete(ctx, updateStrategy)).Should(SatisfyAny(Succeed(), utils.NotFoundMatcher{}))
+		updateStrategy = nil
+
+		By("Deleting the clusterResourceSnapshot")
+		Expect(k8sClient.Delete(ctx, resourceSnapshot)).Should(SatisfyAny(Succeed(), utils.NotFoundMatcher{}))
+		resourceSnapshot = nil
+
+		By("Deleting the clusterResourceOverride")
+		Expect(k8sClient.Delete(ctx, clusterResourceOverride)).Should(SatisfyAny(Succeed(), utils.NotFoundMatcher{}))
+		clusterResourceOverride = nil
+	})
+
+	Context("Test validateCRP", func() {
+		It("Should fail to validate if the CRP is not found", func() {
+			By("Creating a new clusterStagedUpdateRun")
+			Expect(k8sClient.Create(ctx, updateRun)).To(Succeed())
+
+			By("Validating the initialization succeeded")
+			want := generateSucceededInitializationStatus(crp, updateRun, policySnapshot, updateStrategy, clusterResourceOverride)
+			validateSucceededInitializationStatus(ctx, updateRun, want)
```

Comment:
seems that those can be extracted out to the BeforeEach inside this context?

Reply:
Yea, good catch! Fixed.